### PR TITLE
Consume split TokenDetectionsController and CollectibleDetectionController (previously AssetDetectionController)

### PR DESCRIPTION
**Pull request by adonesky1**
This PR makes required changes to consume the split TokenDetectionsController and CollectibleDetectionController (previously AssetDetectionController) in v19.0.0 of @metamask/controllers.

And adapts to use v20.0.0 by adding a call to start polling immediately after instantiating each DetectionController.

### CODE REVIEW

**Comment by adonesky1** [comment]
No idea why CLA is failing here... I've opened and merged PRs into this repo before? The logs in the action don't really provide much clarity?
`Error: Waiting on additional CLA signatures.`

**Comment by Gudahtt** [comment]
The CLA bot is a relatively recent addition.

**Comment by gantunesr** [comment]
Yes, as Mark points out, it's a recent addition. You have to make your organization visibility public or comment the following statement: _I have read the CLA Document and I hereby sign the CLA_

**Review by gantunesr — DISMISSED** [review]
LGTM

**Comment by adonesky1** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by gantunesr** [comment]
@Cal-L spotted some things that should be updated before merging this PR

**Review by cortisiko — CHANGES_REQUESTED** [review]
found an issue: 

My ERC-20 tokens (chainlink & compound)  does not appear after I import my wallet: http://recordit.co/7H7AIeh4b3

Compared to prod: http://recordit.co/x311rIyDT0 

To reproduce:
Launch the app
Import a wallet with eth and ERC-20 tokens

**Comment by adonesky1** [comment]
@cortisiko @gantunesr having alot of of trouble with my mobile environment, could you test again and see if you are still seeing [this issue](https://github.com/MetaMask/metamask-mobile/pull/3380#pullrequestreview-806405016)

File: app/core/Engine.js [hunk 1]
```diff
@@ -173,18 +174,22 @@ class Engine {
...
+					addTokens: tokensController.addTokens.bind(tokensController),
+					getTokensState: () => tokensController.state,
+					getTokenListState: () => tokenListController.state,
+				}),
+				new CollectibleDetectionController({
+					onCollectiblesStateChange: (listener) => collectiblesController.subscribe(listener),
+					onPreferencesStateChange: (listener) => preferencesController.subscribe(listener),
+					onNetworkStateChange: (listener) => networkController.subscribe(listener),
 					getOpenSeaApiKey: () => collectiblesController.openSeaApiKey,
 					getBalancesInSingleCall:
```
Comment (gantunesr) [inline app/core/Engine.js:190]: Looks like `getBalancesInSingleCall` is from `TokenDetectionController`

**Review by gantunesr — CHANGES_REQUESTED** [review]

**Review by gantunesr — COMMENTED** [review]
@adonesky1 The changes I pointed out helps to solve the bug that Curtis got, I also got the same bug but after the changes it was resolved

File: app/core/Engine.js [hunk 2]
```diff
@@ -568,7 +580,6 @@ export default {
 			AddressBookController,
 			AssetsContractController,
 			CollectiblesController,
-			AssetsDetectionController,
 			TokenListController,
 			CurrencyRateController,
 			KeyringController,
```
[the remark this answers is not in the document]
Comment (adonesky1) [reply app/core/Engine.js:585]: Whoops yeah that'll do it 😅 Thanks!

File: app/core/Engine.js (hunk 1 above)
Comment (adonesky1) [reply app/core/Engine.js:190]: Oh you mean it should be passed to the TokenDetectionController not the CollectibleDetectionController. Yes you're right.

**Review by gantunesr — APPROVED** [review]
LGTM

**Comment by cortisiko** [comment]
🌮 🌮 🌮 🌮

**Review by cortisiko — APPROVED** [review]

**Pull request merged**